others/longest_substring_without_repeating_characters.cpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file
+ * @brief Solution for Longest Substring Without Repeating Characters problem.
+ * @details
+ * Problem link: https://leetcode.com/problems/longest-substring-without-repeating-characters/description/
+ * 
+ * Intuition:
+ * The intuition is straightforward and simple. We track the frequency of characters.
+ * Since we can't use a string to track the longest substring without repeating characters
+ * efficiently (as removing a character from the front of a string isn't O(1)),
+ * we optimize the solution using a deque approach.
+ *
+ * Approach:
+ * 1) Initialize an unordered_map to track the frequency of characters.
+ * 2) Use a deque for pushing characters, and update the result deque (`res`) with the current deque (`temp`) 
+ *    whenever we find a longer substring.
+ * 3) Use a while loop to reduce the frequency from the front, incrementing `i`, 
+ *    and removing characters from the `temp` deque as we no longer need them.
+ * 4) Return `res.size()` as we are interested in the length of the longest substring.
+ * 
+ * Time Complexity: O(N)
+ * Space Complexity: O(N)
+ * 
+ * Examples:
+ * - testcase1: s = "abcabcbb", output = 3.
+ * - testcase2: s = "bbbbb", output = 1.
+ * - testcase3: s = "pwwkew", output = 3.
+ * 
+ * I hope this helps to understand.
+ * Thank you!
+ **/
+
+// ----------------- Header files ----------------------------------
+#include <iostream>        // for input and output read/write.
+#include <unordered_map>   // to use it for character frequency.
+#include <deque>           // for push and pop operations at O(1) time.
+#include <string>          // for taking string as input.
+
+using namespace std;       // using the namespace standard to reduce redundant usage of std::
```

Comment:
```suggestion
```
using namespace std is bad practice and hence not allowed on this repository.

Reply:
yeah I have removed the namespace std.